### PR TITLE
Change: Hide buttons in Found Town window that can't be used.

### PR DESCRIPTION
**Pull request by PeterN**
## Motivation / Problem

I thought there was an existing issue for this but I couldn't find one with a search.

When allowed to found towns in game, some buttons are disabled as they are only available in the scenario editor.

image

## Description

Instead of disabling these buttons, completely hide them when in game.

image

## Checklist for review

Some things are not automated, and forgotten often. This list is a reminder for the reviewers.
* The bug fix is important enough to be backported? (label: 'backport requested')
* This PR touches english.txt or translations? Check the [guidelines](https://github.com/OpenTTD/OpenTTD/blob/master/docs/eints.md)
* This PR affects the save game format? (label 'savegame upgrade')
* This PR affects the GS/AI API? (label 'needs review: Script API')
    * ai_changelog.hpp, game_changelog.hpp need updating.
    * The compatibility wrappers (compat_*.nut) need updating.
* This PR affects the NewGRF API? (label 'needs review: NewGRF')
    * newgrf_debug_data.h may need updating.
    * [PR must be added to API tracker](https://wiki.openttd.org/en/Development/NewGRF/Specification%20Status)

### CODE REVIEW

**Comment by 2TallTyler** [comment]
I think you're looking for #10287.

File: src/town_gui.cpp [hunk 1]
```diff
@@ -1194,10 +1218,8 @@ struct FoundTownWindow : Window {
 	void UpdateButtons(bool check_availability)
 	{
 		if (check_availability && _game_mode != GM_EDITOR) {
-			this->SetWidgetsDisabledState(true, WID_TF_RANDOM_TOWN, WID_TF_MANY_RANDOM_TOWNS, WID_TF_LOAD_FROM_FILE, WID_TF_EXPAND_ALL_TOWNS, WID_TF_SIZE_LARGE);
-			this->SetWidgetsDisabledState(_settings_game.economy.found_town != TF_CUSTOM_LAYOUT,
-					WID_TF_LAYOUT_ORIGINAL, WID_TF_LAYOUT_BETTER, WID_TF_LAYOUT_GRID2, WID_TF_LAYOUT_GRID3, WID_TF_LAYOUT_RANDOM);
-			if (_settings_game.economy.found_town != TF_CUSTOM_LAYOUT) town_layout = _settings_game.economy.town_layout;
+			if (_game_mode != GM_EDITOR && _settings_game.economy.found_town != TF_CUSTOM_LAYOUT) town_layout = _settings_game.economy.town_layout;
```
Comment (2TallTyler) [inline src/town_gui.cpp]: We already check the game mode on L1220, although maybe we should ReInit just based on `check_availability`?

```suggestion
			if (_game_mode != GM_EDITOR && _settings_game.economy.found_town != TF_CUSTOM_LAYOUT) this->town_layout = _settings_game.economy.town_layout;
```

File: src/town_gui.cpp [hunk 2]
```diff
@@ -1177,6 +1187,20 @@ struct FoundTownWindow : Window {
 		this->UpdateButtons(true);
 	}
 
+	void OnInit() override
+	{
+		this->GetWidget<NWidgetStacked>(WID_TF_TOWN_ACTION_SEL)->SetDisplayedPlane(0);
+		this->GetWidget<NWidgetStacked>(WID_TF_ROAD_LAYOUT_SEL)->SetDisplayedPlane(0);
```
Comment (2TallTyler) [inline src/town_gui.cpp]: Is there a reason to always set these, instead of having an `else` for the editor check below?

Comment (PeterN) [reply src/town_gui.cpp]: In fact because it's not possible to switch game modes while leaving this window open, that wasn't necessary at all.

**Review by 2TallTyler — APPROVED** [review]

**Pull request merged**